src/google/cloud/ndb/query.py

```diff
@@ -99,6 +102,40 @@ def __init__(self, *args, **kwargs):
         raise NotImplementedError
 
 
+class PropertyOrder(object):
+    """The sort order for a property name, to be used when ordering the
+       results of a query.
+
+       Args:
+           name (str): The name of the model property to use for ordering.
+           direction (str): The sort direction. One of "ascending" (default) or
+               "descending".
+
+        Raises:
+            TypeError: if ``direction`` is not "ascending" or "descending".
+
+    """
+
+    __slots__ = ["name", "direction"]
+
+    def __init__(self, name, direction="ascending"):
+        self.name = name
+        if direction != "ascending" and direction != "descending":
+            raise TypeError("Direction must be 'ascending' or 'descending'")
+        self.direction = direction
+
+    def __repr__(self):
+        return "PropertyOrder(name='{}', direction='{}')".format(
+            self.name, self.direction
+        )
+
+    def __neg__(self):
+        direction = (
+            self.direction == "ascending" and "descending" or "ascending"
```

Comment:
This can't possibly work. Writing a test that flips an expression from descending to ascending should expose this error.

Instead of short circuiting, let's prefer a conditional/ternary expression. ("this" if cond else "that")

Using strings is fine. If we used a boolean, though, and called it "reverse" it would map a little better to the binary state we're modelling here, and would let you do: self.direction = not self.direction. Something to consider, but you decide which you prefer.

Reply:
It does work (really), but I see your point about conditional expressions. Old habits die hard.

Anyway, I decided to use reverse instead, so this is a moot point.

Reply:
Ah, you're right, sorry. I was reading assignment instead of equality.